src/components/tooltip/Marker.js

```diff
@@ -321,37 +321,60 @@ class MarkerTooltipContents extends React.PureComponent<Props> {
           break;
         }
         case 'CompositorScreenshot': {
-          const { width, height } = computeScreenshotSize(
-            data,
-            MAXIMUM_IMAGE_SIZE
-          );
-          details.push(
-            <TooltipDetail label="Image" key="CompositorScreenshot-image">
-              <img
-                className="tooltipScreenshotImg"
-                src={thread.stringTable.getString(data.url)}
-                style={{
-                  width,
-                  height,
-                }}
-              />
-            </TooltipDetail>,
-            <TooltipDetail
-              label="Window Size"
-              key="CompositorScreenshot-window size"
-            >
-              <>
-                {data.windowWidth}px × {data.windowHeight}px
-              </>
-            </TooltipDetail>,
-            <TooltipDetail
-              label="Description"
-              key="CompositorScreenshot-description"
-            >
-              This marker spans the time between each composite of a window and
-              shows the window contents during that time.
-            </TooltipDetail>
-          );
+          if (data.url !== undefined) {
+            const { width, height } = computeScreenshotSize(
+              data,
+              MAXIMUM_IMAGE_SIZE
+            );
+            details.push(
+              <TooltipDetail label="Image" key="CompositorScreenshot-image">
+                <img
+                  className="tooltipScreenshotImg"
+                  src={thread.stringTable.getString(data.url)}
+                  style={{
+                    width,
+                    height,
+                  }}
+                />
+              </TooltipDetail>,
+              <TooltipDetail
+                label="Window Size"
+                key="CompositorScreenshot-window size"
+              >
+                <>
+                  {data.windowWidth}px × {data.windowHeight}px
+                </>
+              </TooltipDetail>,
+              <TooltipDetail
+                label="Description"
+                key="CompositorScreenshot-description"
+              >
+                This marker spans the time between each composite of a window
+                and shows the window contents during that time.
+              </TooltipDetail>,
+              <TooltipDetail
+                label="Window ID"
+                key="CompositorScreenshot-window id"
+              >
+                {data.windowID}
```

Comment:
I wish we could identify the windows better instead of a number. But we need to change the backend to make it happen.